webcomponents/vectortile-view-component/src/app/feature-view/feature-view.component.ts

```diff
@@ -0,0 +1,285 @@
+        this.features = data
+        this.loadfeatures(this.features)
+        this.loadbackground()
+        this.adddragbox()
+        this.addFeatureEmit()
+      })
+  }
+
+  ngOnChanges(changes: NgChanges<FeatureViewComponent>) {
+    if (
+      changes.itemsUrl?.previousValue !== changes.itemsUrl?.currentValue ||
+      changes.projection.previousValue !== changes.projection.currentValue
+    ) {
+      if (changes.itemsUrl?.currentValue) {
+        this.init()
+      }
+    }
+  }
+
+  loadbackground() {
+    switch (this.backgroundMap) {
+      case 'OSM': {
+        const osm = new TileLayer({
+          source: new OSM(),
+        })
+        this.map.addLayer(osm)
+        return
+      }
+      case 'BRT': {
+        this.map.addLayer(this.brtLayer())
+        return
+      }
+
+      default: {
+        exhaustiveGuard(this.backgroundMap)
+        return
+      }
+    }
+  }
+
+  loadfeatures(features: Feature<Geometry>[]) {
+    const vsource = new VectorSource({
+      //features: new GeoJSON().readFeatures(this.featureCollectionGeoJSON, { featureProjection: this.projection  }),
+      features: features,
+    })
+
+    this.map.addLayer(
+      new VectorLayer({
+        source: vsource,
+        style: this.getstyle(),
+        zIndex: 10,
+      })
+    )
+    const ext = vsource.getExtent()
+    if (features.length < 3) {
+      this.setViewExtent(ext, 10)
+    } else {
+      this.setViewExtent(ext, 1.05)
+    }
+
+    return ext
+  }
+
+  getstyle() {
+    const fill = new Fill({
+      color: 'rgba(0,0,255)',
+    })
+    const stroke = new Stroke({
+      color: '#3399CC',
+      width: 1.25,
+    })
+    const styles = [
+      new Style({
+        image: new Circle({
+          fill: fill,
+          stroke: stroke,
+          radius: 5,
+        }),
+        fill: fill,
+        stroke: stroke,
+      }),
+    ]
+    return styles
+  }
+
+  setViewExtent(extent: Extent, scale: number) {
+    const view = new View({})
+    const fitOptions: FitOptions = {
+      size: this.map.getSize(),
+    }
+    const geom = fromExtent(extent)
+    geom.scale(scale)
+    view.fit(geom, fitOptions)
+    this.map.setView(view)
+  }
+
+  brtLayer() {
+    const projectionExtent = [-285401.92, 22598.08, 595401.9199999999, 903401.9199999999]
+    const projection = new Projection({
+      code: 'EPSG:28992',
+      units: 'm',
+      extent: projectionExtent,
+    })
+    const resolutions = [3440.64, 1720.32, 860.16, 430.08, 215.04, 107.52, 53.76, 26.88, 13.44, 6.72, 3.36, 1.68, 0.84, 0.42, 0.21]
+    //const size = ol.extent.getWidth(projectionExtent) / 256
+
+    const matrixIds = []
+    for (let i = 0; i < resolutions.length; ++i) {
+      matrixIds[i] = 'EPSG:28992:' + i
+    }
+    return new Tile({
+      source: new WMTSSource({
+        attributions: 'Kaartgegevens: &copy; <a href="https://www.kadaster.nl">Kadaster</a>',
+        url: 'https://service.pdok.nl/brt/achtergrondkaart/wmts/v2_0?',
```

Comment:
kan deze url naar environment.ts? Dat het expliciet een configuratie setting wordt aangezien deze url ook kan wijzigen in de toekomst.

Reply:
done